set-output: Replace set-ouput usage

batbattur opened the pull request:
## Summary
Since this is the only use set-output used in this action, we can directly set `$GITHUB_OUTPUT` to the result.

## QA 
Make sure the render-template action still works properly.
Might be easier to test this after merging this pull. 
  - We can update the sha in the workflow and create a test pull request. https://github.com/iFixit/ifixit/blob/37a8c72dcd83daa3ca6748667488d654002a95f2/.github/workflows/E2E.yml#L106 
  - Then check the webpack comment. 


Connects: https://github.com/iFixit/ifixit/issues/45080

batbattur committed:
Replace set-ouput usage

Since this is the only use set-ouput used in this action, we can directly set $GITHUB_OUTPUT

batbattur committed:
Update release.yml

mlahargou commented on main.go:
```diff
@@ -56,7 +56,7 @@ func run() error {
 	}
 
-	fmt.Printf("::set-output name=result::%s", escape(output))
+	os.Setenv("GITHUB_OUTPUT", fmt.Sprintf("result=%s", escape(output)))
```
Hmm, does this overwrite GITHUB_OUTPUT? Don't we want to just append to it?

batbattur replied:
> Hmm, does this overwrite GITHUB_OUTPUT? Don't we want to just append to it?

Yes, this overwrites the `GITHUB_OUTPUT`. Since we set it only in one place, I decided to overwrite it. 

I'll look into how to append to it so we don't accidentally overwrite other existing env vars.

batbattur replied:
Okay, looks like this works. I tested it here https://go.dev/play/p/urtPbvAYgH1:

```suggestion
	os.Setenv("GITHUB_OUTPUT", fmt.Sprintln(os.Getenv("GITHUB_OUTPUT"))+fmt.Sprintf("result=%s", escape(output)))
```

mlahargou commented:
We should also make an issue on the upstream repo. Can you do that @batbattur ?

batbattur commented:
dev_block 👍

batbattur committed:
Append to GITHUB_OUPUT

Instead of overriding the output, append to it

batbattur commented:
un_dev_block 👍

batbattur commented on main.go:
```diff
@@ -56,7 +56,7 @@ func run() error {
 	}
 
-	fmt.Printf("::set-output name=result::%s", escape(output))
+	os.Setenv("GITHUB_OUTPUT", fmt.Sprintln(os.Getenv("GITHUB_OUTPUT"))+fmt.Sprintf("result=%s", escape(output)))
```
If by `$GITHUB_OUTPUT` is null, we end up saving the env variable with extra line at the beginning, therefore had to do the following:

```suggestion
	if os.Getenv("GITHUB_OUTPUT") != "" {
		os.Setenv("GITHUB_OUTPUT", fmt.Sprintln(os.Getenv("GITHUB_OUTPUT"))+fmt.Sprintf("result=%s", escape(output)))
	} else {
		os.Setenv("GITHUB_OUTPUT", fmt.Sprintf("result=%s", escape(output)))
	}
```

batbattur replied:
Tested it here https://go.dev/play/p/z60cVkxBX6K

batbattur committed:
Update GITHUB_OUTPUT

If by $GITHUB_OUTPUT is null, we end up saving the env variable with extra line at the beginning, therefore check if it's nil.

batbattur commented:
> We should also make an issue on the upstream repo. Can you do that @batbattur ?

Made the issue here https://github.com/chuhlomin/render-template/issues/7

mlahargou reviewed (approved):
CR 📱

batbattur commented:
@ardelato @mlahargou 

I forgot that this action still uses the author's docker image, so the changed code in here won't matter unless the image gets updated. Just found it here when testing in iFixit/ifixit: https://github.com/iFixit/ifixit/pull/45162#issuecomment-1282901897

ardelato commented:
Oooof, that's actually a bigger problem if we are stilling using an external docker image even though we forked the repo.

ardelato commented:
I am honestly thinking we just create our own script to escape multiline comments in GH.

batbattur commented:
> I am honestly thinking we just create our own script to escape multiline comments in GH.

Created our own script in https://github.com/iFixit/ifixit/pull/45162, and dropped the use of `iFixit/render-template` action.
